share/eds/p2p/client.go

```diff
@@ -0,0 +1,161 @@
+package p2p
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/eds"
+	p2p_pb "github.com/celestiaorg/celestia-node/share/eds/p2p/pb"
+
+	"github.com/celestiaorg/rsmt2d"
+)
+
+var log = logging.Logger("shrex/eds")
+
+var errNoMorePeers = errors.New("all peers returned invalid responses")
+
+// TODO(@distractedm1nd): make version suffix configurable
```

Comment:
We should add the suffix in this PR to minimize tech debt. Also, If we start implementing new patterns on existing code, the new code should integrate those while it developed, not later.